src/java/org/apache/cassandra/hints/HintsService.java

```diff
@@ -390,6 +390,23 @@ public Future transferHints(Supplier<UUID> hostIdSupplier)
         return dispatchExecutor.transfer(catalog, hostIdSupplier);
     }
 
+    /**
+     * Get the earliest hint written for a particular node,
+     * @param hostID UUID of the node to check it's hints.
+     * @return Earliest hint as per unix time or Long.MIN_VALUE if hostID is null
+     */
+    public long getEarliestHintForNode(UUID hostID)
+    {
+        if (hostID == null)
+            return Long.MIN_VALUE;
```

Comment:
```suggestion
        Preconditions.checkNotNull(hostId);
```

There's no reason to accept a null argument here. And is already being guarded against here: https://github.com/apache/cassandra/pull/1152/files#diff-4f6615c703489b87cafd23c0dc5a4edc4af27c33aba172870b30e38a7d8faa1dR2172

Reply:
That's not completely true, there is a test called "WriteCallbackInfoTest" which as a by-product of doing what it does puts null in there. Without that check, it would throw NPE at some places.

Reply:
this is a brand new method, there's no need for it to accept a null.  #KISS